lib/GeoExt/data/PrintProviderBase.js

```diff
@@ -0,0 +1,797 @@
+     *  (in milliseconds). Default of 30 seconds. Has no effect if ``method``
+     *  is set to ``GET``.
+     */
+    timeout: 30000,
+
+    /** api: property[customParams]
+     *  ``Object`` Key-value pairs of custom data to be sent to the print
+     *  service. Optional. This is e.g. useful for complex layout definitions
+     *  on the server side that require additional parameters.
+     */
+    customParams: null,
+
+    /** api: config[baseParams]
+     *  ``Object`` Key-value pairs of base params to be add to every
+     *  request to the service. Optional.
+     */
+
+    /** api: property[scales]
+     *  ``Ext.data.JsonStore`` read-only. A store representing the scales
+     *  available.
+     *
+     *  Fields of records in this store:
+     *
+     *  * name - ``String`` the name of the scale
+     *  * value - ``Float`` the scale denominator
+     */
+    scales: null,
+
+    /** api: property[dpis]
+     *  ``Ext.data.JsonStore`` read-only. A store representing the dpis
+     *  available.
+     *
+     *  Fields of records in this store:
+     *
+     *  * name - ``String`` the name of the dpi
+     *  * value - ``Float`` the dots per inch
+     */
+    dpis: null,
+
+    /** api: property[outputFormats]
+     *  ``Ext.data.JsonStore`` read-only. A store representing the output formats
+     *  available.
+     *
+     *  Fields of the output formats in this store:
+     *
+     *  * name - ``String`` the name of the output format
+     */
+    outputFormats: null,
+
+    /** api: property[outputFormatsEnabled]
+     *  ``Boolean`` read-only. Should outputFormats be populated and used?
+     *  Default value is 'False'
+     */
+    outputFormatsEnabled: false,
+
+    /** api: property[layouts]
+     *  ``Ext.data.JsonStore`` read-only. A store representing the layouts
+     *  available.
+     *
+     *  Fields of records in this store:
+     *
+     *  * name - ``String`` the name of the layout
+     *  * size - ``Object`` width and height of the map in points
+     *  * rotation - ``Boolean`` indicates if rotation is supported
+     */
+    layouts: null,
+
+    /** api: property[dpi]
+     *  ``Ext.data.Record`` the record for the currently used resolution.
+     *  Read-only, use ``setDpi`` to set the value.
+     */
+    dpi: null,
+
+    /** api: property[layout]
+     *  ``Ext.data.Record`` the record of the currently used layout. Read-only,
+     *  use ``setLayout`` to set the value.
+     */
+    layout: null,
+
+    /** api: property[outputFormat]
+     *  ``Ext.data.Record`` the record of the currently used output format. Read-only,
+     *  use ``setOutputFormat`` to set the value.
+     */
+    outputFormat: null,
+
+    /** api: property[defaultOutputFormatName]
+     *  ``String`` the name of the default output format.
+     */
+    defaultOutputFormatName: 'pdf',
+
+    /** private: attribute[attributes]
+     *  ``Array`` all the current attributes
+     */
+    attributes: [],
```

Comment:
should we be instantiating an array on the prototype? I thought this was unsafe?